internal/tofu/node_resource_deposed.go

```diff
@@ -329,9 +329,7 @@ func (n *NodeDestroyDeposedResourceInstanceObject) Execute(ctx context.Context,
 		return diags
 	}
 
-	diags = diags.Append(n.postApplyHook(evalCtx, state, diags.Err()))
-
-	return diags.Append(updateStateHook(evalCtx))
```

Comment:
The call above to `writeResourceInstanceState` calls `SetResourceInstanceDeposed` which can remove the instance from the state and eventually can actually remove the whole resource from the state.
Are we sure that by removing the `updateState` from here will have no side effects?
Feels that in this case we might end up in some changes not being reflected in the state snapshot.

Please correct me if my understanding is wrong here.

Reply:
Good catch, let me address this.